### PR TITLE
allow more customisation in cardinality, fuzziness and range

### PR DESCRIPTION
**Pull request by aspacca**

### CODE REVIEW

**Review by endorama — APPROVED** [review]
Small typo and a question. Overall :+1:

File: README.md [hunk 1]
```diff
@@ -159,22 +161,40 @@ And the following config file content:
 - name: AccountID
   value: 123456789012
 - name: InterfaceID
-  cardinality: 10
+  cardinality:
```
[the remark this answers is not in the document]
Comment (aspacca) [reply README.md:164]: Yes, I would add in a separate PR. what about [yours for docs](https://github.com/elastic/elastic-integration-corpus-generator-tool/pull/45)?

File: pkg/genlib/generator_with_custom_template_test.go [hunk 2]
```diff
@@ -228,10 +218,26 @@ func test_CardinalityTWithCustomTemplate[T any](t *testing.T, ty string) {
...
+			cardinalityDenominator /= cardinality
+		}
+
+		rangeTrailing := ""
+		if ty == FieldTypeFloat {
+			rangeTrailing = "."
+		}
+
+		rangeMin := rand.Intn(100)
+		rangeMax := rand.Intn(10000-rangeMin) + rangeMin
```
[the remark this answers is not in the document]
Comment (aspacca) [reply pkg/genlib/generator_with_custom_template_test.go:235]: as you know, I prefer to have some fuzziness in tests, to catch potential hidden bugs ;)

[the remark this answers is not in the document]
Comment (endorama) [reply pkg/genlib/generator_with_custom_template_test.go:235]: I'm not sure this test is actually behaving like that. Without updating the `seed` the generated values will not change. ([Seed docs](https://pkg.go.dev/math/rand#Seed), it behaves like always using `Seed(1)`)

See for a playground: https://go.dev/play/p/vCLM9roYzuw

From my understanding without updating the seed (I'm not sure if this happens somewhere, but searching for `Seed` or `rand.Seed` in this repo given 0 results) this code behave like using a constant value.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (endorama) [reply README.md:164]: ok on doing in a separate PR. I will rebase my doc and update it once this is merged

File: pkg/genlib/generator_with_custom_template_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (aspacca) [reply pkg/genlib/generator_with_custom_template_test.go:235]: > From my understanding without updating the seed (I'm not sure if this happens somewhere, but searching for `Seed` or `rand.Seed` in this repo given 0 results) this code behave like using a constant value.

yes, I don't update the seed, indeed. but that's not needed: every call to `rand.Intn` does not generate a constant value, rather, multiple calls with the same seed generate always the same sequence: 
> Seed values that have the same remainder when divided by 2³¹-1 generate the same pseudo-random sequence

I suppose that once the sequence is completely generated it will start over. I have to look how long is the sequence (int64 size?) but we are far from filling it :)

so basically we test always with the same sequence, indeed, that's why I mentioned that "we can also have a fixed list of ranges for every cardinality" :)

or we can use the current unix timestamp as seed, reducing the number of identical sequences between each tests run
